[31기 남용현] MODIFY: 공통 nav 수정사항 반영 완료

sunnyfterrain opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [x] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [x] 컨벤션 수정

<br />

## :: 구현 목표
- 공통 nav 제작

<br />

## :: 구현 사항 설명 (
1. navBar 메뉴 제작
- 기능구현 없이 단순 정적인 nav ui 구현

<br />

## :: 성장 포인트 

<br />

## :: 기타 질문 및 특이 사항

sunnyfterrain committed:
ADD: nav 1차 제작완료
- 기능구현 및 동적 제외
- 정적인 ui 만 구현 완료

sunnyfterrain committed:
Modify: nav 글씨 간격 수정

sunnyfterrain committed:
Modify: font 및 position 수정

yeonuk-hwang commented on jsconfig.json:
```diff
@@ -1,6 +0,0 @@
-    "baseUrl": "src"
-  },
-  "include": ["src"]
-}
```
이부분은 왜 삭제하신건가요?

sunnyfterrain replied:
저부분을 넣고 scss import를 하면 자꾸 제 vscode가 강제 종료되는 현상이 있어 삭제 했습니다😭

yeonuk-hwang commented on src/components/Header/Header.js:
```diff
@@ -1,7 +1,24 @@
+import './Header.scss';
 
 const Header = () => {
-  return <div>Header</div>;
```
전체적으로 최상위 태그의 className을 컴포넌트 이름과 동일하게 부여한 뒤 그 className하위로 네스팅 하는 부분이 적용안되어있습니다. 이렇게 되면 동일한 className을 가진 다른 컴포넌트의 요소들에게도 영향을 미칠 수 있습니다.
**최상위 태그의 className을 컴포넌트 이름과 동일하게 부여한 뒤 그 className하위로 네스팅** 적용해주세요

예시)

```jsx
const ProductDetail = () => {

	return(
		<section className="productDetial">
			~~~~
		</section>
	)
}
```

sunnyfterrain replied:
최상위 태그를 컴포넌트 이름과 동일하게 부여하였습니다.

yeonuk-hwang commented on src/components/Header/Header.scss:
```diff
@@ -0,0 +1,51 @@
+@import '../../styles/variables.scss';
+
+header {
```
위에 드린 코멘트와 연관되는 내용으로 이렇게 되면 프로젝트 내에 모든 header tag에 이 스타일이 적용되어 버리겠네요

sunnyfterrain replied:
기본 태그 선택자 대신 모두 클래스네임을 적용하였습니다.

yeonuk-hwang commented on src/components/Header/Header.scss:
```diff
@@ -0,0 +1,51 @@
+  }
+
+  .navTitle {
+    display: flex;
+    flex-direction: column;
```
필요한 속성인가요? 아래에 해당되는 태그들이 다 block 이라서 굳이 flex, column 안해도 세로 쌓임으로 동작할 것 같습니다!

sunnyfterrain replied:
해당 속성은 필요가 없어 삭제하였습니다.

yeonuk-hwang commented on src/components/Header/Header.scss:
```diff
@@ -0,0 +1,51 @@
+    flex-direction: column;
+    margin-right: 95px;
+
+    h3 {
```
- div 태그 처럼 빈번히 사용되는 태그로 스타일을 주는 건 depth에 무관하게 적용되어 의도치 않은 스타일링이 될 수 있습니다. 클래스 셀렉터(ex. `.foo`)를 사용해주세요.
- tag selector는 해당 요소 뿐만 아니라 동일한 태그로 선언된 다른 요소들에게도 영향을 미칠 여지가 너무 많기 때문에 추후 유지보수에 악영향을 미칠 수 있습니다.
- 스타일링을 줄 때는 요소마다 className을 부여하셔서 스타일링을 주시는게 좋습니다.

sunnyfterrain replied:
기본 태그 선택자를 없애고 모두 className으로 변경하였습니다.

yeonuk-hwang commented on src/pages/Main/Main.js:
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
+import Header from '../../components/Header/Header';
+import './Main.scss';
```
main.scss가 empty인데 import 한 이유가 있나요?

sunnyfterrain replied:
삭제하였습니다!

yeonuk-hwang commented on src/pages/Main/Main.js:
```diff
@@ -1,11 +1,12 @@
-      Main
-      <div className="aa">d푸룯츠</div>
-    </div>
+    <main className="container">
```
container라는 className은 어디서 스타일링을 해주고 있을까요?!

sunnyfterrain replied:
현재는 main에 작업한 사항이 없어 삭제하였습니다.

yeonuk-hwang reviewed (changes requested):
고생하셨습니다 용현님!
- PR 제목 양식 맞춰주시고
- 라벨도 잘 붙여주세요 :)

sunnyfterrain committed:
Modify: 클래스네임, 불필요한 스타일링 등 수정사항
- 기본 선택자 대신 클래스네임 부여
- 최상위 부모태그 클래스네임과 컴포넌트명 일치
- 불필요한 스타일속성 제거
- 불필요한 import 및 파일 제거

sunnyfterrain committed:
Modify: variables, common, jsconfig 수정

201steve reviewed (approved):
-
